test/metric_value_benchmark/statsd_test.go

```diff
@@ -44,62 +68,123 @@ func (t *StatsdTestRunner) GetAgentConfigFileName() string {
 }
 
 func (t *StatsdTestRunner) GetAgentRunDuration() time.Duration {
-	return time.Minute
+	return 3*time.Minute
 }
 
 func (t *StatsdTestRunner) SetupAfterAgentRun() error {
-	// EC2 Image Builder creates a bash script that sends statsd format to cwagent at port 8125
-	// The bash script is at /etc/statsd.sh
-	//    for times in  {1..3}
-	//    do
-	//      echo "statsd.counter:1|c" | nc -w 1 -u 127.0.0.1 8125
-	//      sleep 60
-	//    done
-	startStatsdCommand := []string{
-		"sudo bash /etc/statsd.sh",
+	// For each test run we want a unique metric name.
+	// Just in case 2 different people run this test in parallel, using the
+	// same AWS account...
+	// Populate a temp map with the new metric names.
+	suffix := fmt.Sprint(time.Now().UnixNano())
+	newMap := map[string]metricInfo{}
+	for k, v := range metricMap {
+		newKey := k + suffix
+		newMap[newKey] = v
 	}
-
-	return common.RunCommands(startStatsdCommand)
+	// Replace global map with the temp one.
+	metricMap = newMap
+
+	// Send unique metrics each second.
+	// Expect agent to collect every 5 seconds.
+	// Expect agent to aggregate collections into 30 second buckets.
```

Comment:
Please verify my math makes sense to you.
And check it against the JSON config.